### PR TITLE
Use triggerv2 controller in shared main

### PR DESCRIPTION
**Pull request by Cali0707**
As we move towards scaling triggers with KEDA, we first need to migrate triggers to use consumergroups.

## Proposed Changes

- Switch triggers to use the triggerv2 controller in shared main.
- Update the deployments to use the `contract-resources` configmap, and for the broker-dispatcher to be deployed as a statefulset
-

**Release Note**

```release-note
Triggers for Kafka brokers (but not namespaced brokers) now use consumergroups internally.
```

**Docs**

### CODE REVIEW

**Comment by Cali0707** [comment]
@pierDipi any ideas why the kafka-broker-dispatcher statefulset for the namespaced brokers has replicas:0 with these changes?

**Comment by Cali0707** [comment]
/cc @pierDipi 
This is ready for review

Per discussion on slack, this shouldn't be merged until after 1.13 branch cut
/hold

File: control-plane/pkg/reconciler/trigger/v2/triggerv2.go [hunk 1]
```diff
@@ -60,25 +63,28 @@ const (
...
+	ConsumerGroupLister  internalslst.ConsumerGroupLister
+	InternalsClient      internalsclient.Interface
+	SecretLister         corelisters.SecretLister
+	KubeClient           kubernetes.Interface
+	KafkaFeatureFlags    *apisconfig.KafkaFeatureFlags
+	FeatureStore         feature.Store
 }
 
 func (r *Reconciler) ReconcileKind(ctx context.Context, trigger *eventing.Trigger) reconciler.Event {
-	trigger.Status.MarkOIDCIdentityCreatedNotSupported()
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/trigger/v2/triggerv2.go:75]: where do we set this condition now?

Comment (pierDipi) [reply control-plane/pkg/reconciler/trigger/v2/triggerv2.go:75]: in `SetupOIDCServiceAccount` ?

Comment (Cali0707) [reply control-plane/pkg/reconciler/trigger/v2/triggerv2.go:75]: Yeah, `SetupOIDCServiceAccount` should do that, as far as I know

File: control-plane/cmd/post-install/kafka_broker_deployment_deleter.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+		if err := k.deleteDeployment(ctx, deployment); err != nil {
+			return fmt.Errorf("failed to delete deployment %s: %v", deployment, err)
+		}
+	}
+
+	return nil
+}
+
+func (k *kafkaDeploymentDeleter) deleteDeployment(ctx context.Context, deploymentName string) error {
+	err := k.waiteStatefulSetExists(ctx, deploymentName)
```
Comment (pierDipi) [inline control-plane/cmd/post-install/kafka_broker_deployment_deleter.go:51]: typo `waite....` unless it's a word I don't know about

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

One nit, we can fix it later

**Comment by Cali0707** [comment]
@pierDipi can I unhold this PR?

**Comment by Cali0707** [comment]
It looks like the hold was originally because this PR was on hold because of the 1.13 branch cut, I'm unholding now since we are a month away from the branch cut for 1.14

/unhold

**Comment by Cali0707** [comment]
/cc @pierDipi 

Would you mind taking a look at my most recent commit? I changed the namespaced_controller for the triggers to use a filtered global resync, and to resync the triggers instead of the brokers. This seems to have fixed the namespaced broker tests, but I'm not sure if there was a reason in the past for en-queuing the namespaced brokers instead of triggers when doing a global resync of the triggers

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

File: control-plane/pkg/reconciler/channel/channel.go [hunk 3]
```diff
@@ -477,9 +477,6 @@ func (r *Reconciler) finalizeKind(ctx context.Context, channel *messagingv1beta1
 	}
 
 	authContext, err := security.ResolveAuthContextFromLegacySecret(secret)
-	if err != nil {
-		return fmt.Errorf("failed to resolve auth context: %w", err)
-	}
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/channel/channel.go:480]: Not sure why we dropped this here

**Pull request merged**